src/components/ConditionalFilter/QueryBuilder/QueryBuilder.ts

```diff
@@ -0,0 +1,86 @@
+import { FilterDefinitionResolver } from "../filterDefinitions/FilterDefinitionResolver";
+import {
+  FilterDefinition,
+  FilterQuery,
+  supportsFilterApi,
+  supportsWhereApi,
+} from "../filterDefinitions/types";
+import { FilterContainer, FilterElement } from "../FilterElement";
+import { QueryApiType } from "./types";
+
+export class QueryBuilder<T extends FilterQuery, K extends keyof T = never> {
```

Comment:
It should proably be names "QueryFiltersBuilder" because
- It's actually not building query
- It's abstracting away api details on how fitlers are used by making generic "filters" inside